kore/src/Kore/Strategies/Goal.hs

```diff
@@ -1008,6 +988,15 @@ removalPredicate
           , Pretty.indent 4 (unparse destTerm)
           ]
   where
+    evaluateMultipleExists patt = do
+        let extraElemVariables =
+                getExtraElemVariables configuration patt
+            remainderPattern =
+                Pattern.fromCondition
+                . Conditional.withoutTerm
+                $ (const <$> destination <*> patt)
+        Exists.makeEvaluate
```

Comment:
I think we should use `Exists.simplifyEvaluated` here, so that we don't need the `traverse` + `OrPattern.flatten` combination above. (`traverse` on an `OrPattern` is not a good idea.) Using `simplifyEvaluated`, we can simply fold over the list of variables to be quantified.